codecommit-put-file-action/entrypoint.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -e
+
+# set aws-credentials
+mkdir -p ~/.aws
+touch ~/.aws/credentials
+
+echo "[default]
+aws_access_key_id = ${AWS_ACCESS_KEY_ID}
+aws_secret_access_key = ${AWS_SECRET_ACCESS_KEY}" > ~/.aws/credentials
+
+set +e # May fail
+BRANCH_INFO=`aws codecommit get-branch --repository-name $AWS_CODECOMMIT_REPO_NAME --branch-name $AWS_CODECOMMIT_BRANCH_NAME`
+BRANCH_INFO_STATUS=$?
+set -e
+if [ $BRANCH_INFO_STATUS -ne 0 ]; then
+    # this would be the first commit to this branch
+    BRANCH_COMMIT_ID=''
+else
+    # this would be the head commit on this branch
+    BRANCH_COMMIT_ID=`echo $BRANCH_INFO | jq -r '.branch.commitId'`
+fi
+
+# parent commit flag is required if the repo exists, else no flag is required. (https://docs.aws.amazon.com/cli/latest/reference/codecommit/put-file.html) 
+export PARENT_COMMIT_FLAG=""
+if [ -n "$BRANCH_COMMIT_ID" ]; then
+    PARENT_COMMIT_FLAG="--parent-commit-id=$BRANCH_COMMIT_ID"
+fi
```

Comment:
A parent commit id pointing to the head commit of the branch you want to put-file to is required. [ref](https://docs.aws.amazon.com/cli/latest/reference/codecommit/put-file.html) Hence, it makes sense to not decouple the step of fetching the head commit into a separate generic GH action.

The behavior of continuing to create new branch if the branch does not exist as well as allowing empty commit messages follows from the codecommit put-file behavior and also matches with the current Travis CI workflow. It might make sense to constrain such usage for our use-case but for a generic codecommit-put-file GH action, this behavior should be unconstrained.